testsuite/features/step_definitions/command_steps.rb

```diff
@@ -420,7 +420,11 @@
       sleep checking_rate
     end
   rescue StandardError => e
-    log e.message # It might be that the MU repository is wrong, but we want to continue in any case
+    log e.message
+    # It might be that the MU repository is wrong, but we want to continue in any case
+    unless $build_validation
+      raise ScriptError, "This channel was not fully synced: #{channel}"
```

Comment:
Because it synchronizes channels sequentially, it would be a great enhancement if we will be able to retrieve what channels were not synced. For example we can read from remote source that it consist of 5 channels of that names, if only 3 of them exist in `.../rhn/.../reposync.log`. Can we call back channels that were not make it?

Reply:
Isn't that what the step does? It checks for one channel at a time to be synced and if not finished in the time specified in the TIMEOUT_BY_CHANNEL_NAME for that specific channel, it raised an exception with the name of the channel

Reply:
At least this step has as a scope waiting for a particular channel. For that reason, I would not add extra responsibilities to it.

But your suggestion make more sense in the other changed step `When(/^I wait until all synchronized channels for "([^"]*)" have finished$/) do |os_product_version|` where we waiting for a set of channels.

What we do in that different step, is to rely into a function that checks by the existence of a file that must be placed once the channel is fully synchronized, and we  iterate over all these channels waiting for them to be synchronized (in a order that can't be determined) until all of them are completed.

Reply:
Ok, it makes sense.